core/src/main/java/io/parsingdata/metal/token/Token.java

```diff
@@ -75,6 +75,10 @@ protected Token(final String name, final Encoding encoding) {
 
     protected abstract Optional<ParseState> parseImpl(final Environment environment);
 
+    public boolean isScopeDelimiter() {
```

Comment:
I know we haven't documented the other methods here (like `isLocal()` below), but these really do need proper Javadoc. The meaning of these `boolean`s are now "hidden" in the Javadoc in the classes that need these fields. We could add some short documentation and refer to the other classes for more detailed explanations.

Reply:
Perhaps also adding the rationale behind the choice of when something would be a scope delimiter.

Reply:
Agreed again on rationale, but like I said earlier, we really need more comprehensive documentation for users of the entire library. I would rather focus on that than make lots of Javadoc.